posthog-node/src/posthog-node.ts

```diff
@@ -203,6 +204,69 @@ export class PostHog implements PostHogNodeV1 {
     return response
   }
 
+  async getFeatureFlagPayload(
+    key: string,
+    distinctId: string,
+    matchValue?: string | boolean,
+    options?: {
+      groups?: Record<string, string>
+      personProperties?: Record<string, string>
+      groupProperties?: Record<string, Record<string, string>>
+      onlyEvaluateLocally?: boolean
+      sendFeatureFlagEvents?: boolean
+    }
+  ): Promise<JsonType | undefined> {
+    const { groups, personProperties, groupProperties } = options || {}
+    let { onlyEvaluateLocally, sendFeatureFlagEvents } = options || {}
+    let response = undefined
+
+    // Try to get match value locally if not provided
+    if (!matchValue) {
+      matchValue = await this.getFeatureFlag(key, distinctId, {
+        ...options,
+        onlyEvaluateLocally: true,
```

Comment:
You can simplify this quite a bit by passing in `onlyEvaluateLocally` here. What you really care about is the `matchValue`, so the burden of local evaluation can be completely passed to `getFeatureFlag`

Reply:
Hmm, I made it explicitly local on that call because I don't want to defer to decide endpoint as a backup until the very end of the function. So the flow is meant to be -> try everything local first (get value, then get payload) if that's not possible, then we do reinit the shared client which will trigger the decide endpoint call and then return the payload that's desired. If i just forward `onlyEvaluateLocally`, the decide endpoint will be called on getFeatureFlag but the next statement where it tries to retrieve payload locally could fail and then the decide would get hit again upon shared client logic

RE: `should do getFeatureFlagPayloads without matchValue` this test should show the problem. We want to limit it to 1 call and not accidentally hit 2

Reply:
Not sure I understand.

If matchValue isn't available locally, you'll have to go anyway to decide, yes? And if it's locally evaluable, you wouldn't go to decide anyway.

> If i just forward onlyEvaluateLocally, the decide endpoint will be called on getFeatureFlag but the next statement where it tries to retrieve payload locally could fail and then the decide would get hit again upon shared client logic

Ah, this seems like a bug if it happens? Why would that next statement to retrieve payload locally fail, once we just went to decide?

Reply:
Different explanation: if decide needs to happen, we only want it to happen once so it may as well be the right call that will return the payload corresponding to the flag key. I'm intending for `getFeatureFlag` _and_ `getFeatureFlagPayload` to be the logic that tries to handle verything locally, if none of that flies, then rollover to the decide based call. So, that's why the first `getFeatureFlag` call is only needed for the local evaluation code path. If local is a bust, we're going to make a call anyway so we may as well let the right payload logic call happen and not multiple decides

Reply:
I guess the part I don't still understand is, why do 2 decide calls happen?

Whenever we call `reloadFeatureFlagsAsync`, not only do we update flags, but also all payloads, no matter where it's called from.

Reply:
Summary of call:
The way that poller and decide results are handled are separate. The decide result can be handled better but that's a matter of refactoring that is out of scope for this PR